### ISSUE TITLE
Overnight Oddities - increased XP/level, dailies changed, tasks duplicated

### ISSUE THREAD

**Issue by Atramagus** [HabitRPG/habitica#4622]
My wife was mid level 8 last night before we went to bed. When she woke up in the morning, she had **somehow advanced to mid-level 9, and gotten several new drops**. There's no one else in the apartment.

**None of her dailies that she checked yesterday were unchecked** this morning, but her **streak count was wrong** (supposed to be 14-15, all had dropped to 5 or below), some of her **to-dos were duplicated**. 

When she attempted to reorder to dos, they would automatically reorder themselves, making it hard for her to use it as a task list for the day.

Is there anyway you can check to see if there was something odd going on? She's using Google Chrome on Windows 8.1, as well as the Android app. The Android App hadn't been working for several days, and after this trouble she uninstalled the app from her phone.

**Comment by Alys** [comment]
We would need to know who she is before we can look into this. Please ask her to read this:  

In the website, go to [Settings -> API](https://habitrpg.com/#/options/settings/api) where you'll find a long string of characters called a "User ID". Please copy that to the clipboard and paste it into this issue. Once we have that, we can look at your account to try to work out what happened.

**Comment by Atramagus** [comment]
Sure! User ID: f5a0e1de-6736-480c-9df7-45e6e25b2bf7

Edit: After having her close and delete the android app, I had her check her browser. Still can't sort. To be thorough, I'm having her clear her browser cache and restart chrome, but this will take a few minutes as she needs to save a number of tabs that she has opened for research.

**Comment by Alys** [comment]
**rain1085**: 

For your To-Do sorting problem, please see "Rearranging To-Dos" in the wiki's [To-Dos](http://habitrpg.wikia.com/wiki/To-Dos#Rearranging_To-Dos) page, and the "Clean up" section it refers to. In brief, if you delete your completed To-Dos, your problems with sorting the remaining ones should go away for a few days at least.

Can you tell me please how many new drops you received overnight, which items they were (as best as you can remember), and how you noticed that you'd been given them? For example, did you see the green drop notifications in a browser, or did you just notice that your inventory had more items than before? If the latter, when was the last time you had looked at your inventory?

Before I can look at your tasks in detail, I'd like to have your permission to do so. Can you please add a comment here? Or, if you don't want to create a GitHub account, send me a personal message in HabitRPG, saying that you want me to look at your account to investigate and fix the problems with your tasks (duplication and other odd behaviour). The easiest way to send me a message is to go to [the Tavern](https://habitrpg.com/#/options/groups/tavern), click on my name in the list of moderators (see the orange box highlight in [this screenshot](http://i.gyazo.com/f8b021c1ce1373d73276140a4d0262c3.png)), and then click on the message icon in the screen that appears (see the orange box highlight at the bottom of [this other screenshot](http://i.gyazo.com/d92e1533ed9e5fda112a3e979a7cd1e1.png)).

**Comment by Atramagus** [comment]
The items that appeared were a panda bear egg, tiger egg, (maybe a fox egg) and a blue hatching potion. There was no browser notification, she checked her inventory to feed a pet and all the items were there. She had last looked at her inventory sometime the night before this occurred. 
Unfortunately clearing her cache did not resolve the issue, and her tasks were still rearranging themselves this morning. She just deleted her completed to dos, and we'll see if that helps. I'll have her send you the message soon.

**Comment by crookedneighbor** [comment]
Is she using any third party extensions like the pomodoro(sp?) extension?

I use the todoist-habitprg module, so I regularly have new things appearing in my inventory without notice because the module is checking off tasks I've completed using the api. Sometimes I don't notice I have new food for my pets for days!

Anyway, that's one possible explanation for the new items.

**Comment by Atramagus** [comment]
The only extensions that she has enabled are Evernote Web Clipper, Google Docs, and Hangouts, none of which would interact with the page. They are the same extensions that I have enabled, and I haven't experienced the issue.
Thanks for continuing to investigate! 

```
 On Wednesday, February 4, 2015 9:01 AM, Blade Barringer <notifications@github.com> wrote:
```

 Is she using any third party extensions like the pomodoro(sp?) extension?I use the todoist-habitprg module, so I regularly have new things appearing in my inventory without notice because the module is checking off tasks I've completed using the api. Sometimes I don't notice I have new food for my pets for days!Anyway, that's one possible explanation for the new items. —

**Comment by Atramagus** [comment]
Okay, she should have sent you a message directly through the tavern. Let us know if there is anything else we can do to help you investigate!

**Comment by Atramagus** [comment]
More issues happened overnight:

She had checked off nearly all her dailies last night, received the rewards for doing so. This morning, after refreshing the page so that she could check her dailies, "Read fiction before bed" was left checked, and the streak for it was reset. This appears to be the only one that was messed up.

**Comment by Alys** [comment]
@Atramagus and rain1085: That latter problem of one Daily remaining checked will be due to this bug: https://github.com/HabitRPG/habitrpg/issues/4313. If that keeps happening, there's a workaround described near the top of the page ("What to do if you experience this bug often" heading). I'll be looking at the other problems on the weekend.

**Comment by Atramagus** [comment]
Awesome, definitely appreciated! We'll try the steps listed tonight and every night after for a few days to see if it helps.

**Comment by Atramagus** [comment]
Today's update - we tried the method mentioned in #4313 and there was no oddities when she logged in this morning. We'll keep this up for the next few nights and see if anything further happens.

**Comment by Alys** [comment]
I've been talking to rain1085 by PM and she says that the duplicated tasks have disappeared (which confirms what I saw in the database when I checked her account this morning). 

I think all the problems reported here were due to [this bug](https://github.com/HabitRPG/habitrpg/issues/4313) and since we're tracking it in that issue, I'll close this one. However we can definitely reopen this issue if necessary, and I've told rain1085 that she should let us know if there's any more problems, especially ones that she needs help to fix. Atramagus, thanks for your help with reporting this!
